File: python/Video.py
import yt

from datetime import datetime

class Video:
    def __init__(self, ptr):
        self._ptr = ptr

    def id(self):
        return self._ptr.id

    def title(self):
        return self._ptr.title

    def channel_name(self):
        return self._ptr.channel_name

    def channel_id(self):
        return self._ptr.channel_id

    def published(self):
        return datetime.utcfromtimestamp(self._ptr.published)

    def playable(self, quality):
        return yt.video_get_playable(self._ptr, int(quality))

    def __repr__(self):
        return "{} - {}".format(self.channel_name(), self.title())

    def __del__(self):
        yt.video_free(self._ptr)

File: src/main/mm/Video.cpp
//
// Created by Rhys on 18/02/2022.
//

#include <Video.h>

#include <Config.h>

mm::Video::Video(struct ::Video *ptr) {
    this->ptr = ptr;
}

struct ::Video* mm::Video::cptr() {
    return ptr;
}


std::string mm::Video::id() {
    return std::string(ptr->id);
}

std::string mm::Video::title() {
    return std::string(ptr->title);
}

std::string mm::Video::channel_name() {
    return std::string(ptr->channel_name);
}

std::string mm::Video::channel_id() {
    return std::string(ptr->channel_id);
}

size_t mm::Video::published() {
    return ptr->published;
}

std::string mm::Video::get_playable(mm::Config *conf) {
    return std::string(video_get_playable(ptr, conf->cptr()));
}

mm::Video::~Video() {
    video_free(ptr);
}